From df39ffe7ac712083e5a1b2c7efba71a4d87fb015 Mon Sep 17 00:00:00 2001
From: Adrien Plazas <kekun.plazas@laposte.net>
Date: Wed, 12 Jan 2022 16:12:02 +0100
Subject: [PATCH] updates-section: Ellipsize the button label

This ensures the button don't grow so wide it makes the window pop out
of the screen.
---
 src/gs-updates-section.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/src/gs-updates-section.c b/src/gs-updates-section.c
index 70dbf7f84..0ed284f54 100644
--- a/src/gs-updates-section.c
+++ b/src/gs-updates-section.c
@@ -323,6 +323,8 @@ _update_buttons (GsUpdatesSection *self)
 
 	if (self->kind == GS_UPDATES_SECTION_KIND_OFFLINE_FIRMWARE ||
 	    self->kind == GS_UPDATES_SECTION_KIND_OFFLINE) {
+		GtkWidget *child;
+
 		if (_all_offline_updates_downloaded (self))
 			gtk_stack_set_visible_child_name (GTK_STACK (self->button_stack), "update");
 		else
@@ -332,12 +334,18 @@ _update_buttons (GsUpdatesSection *self)
 		/* TRANSLATORS: This is the button for installing all
 		 * offline updates */
 		gtk_button_set_label (GTK_BUTTON (self->button_update), _("Restart & Update"));
+		child = gtk_bin_get_child (GTK_BIN (self->button_update));
+		gtk_label_set_ellipsize (GTK_LABEL (child), PANGO_ELLIPSIZE_END);
 	} else if (self->kind == GS_UPDATES_SECTION_KIND_ONLINE) {
+		GtkWidget *child;
+
 		gtk_stack_set_visible_child_name (GTK_STACK (self->button_stack), "update");
 		gtk_widget_show (GTK_WIDGET (self->button_stack));
 		/* TRANSLATORS: This is the button for upgrading all
 		 * online-updatable applications */
 		gtk_button_set_label (GTK_BUTTON (self->button_update), _("Update All"));
+		child = gtk_bin_get_child (GTK_BIN (self->button_update));
+		gtk_label_set_ellipsize (GTK_LABEL (child), PANGO_ELLIPSIZE_END);
 	} else {
 		gtk_widget_hide (GTK_WIDGET (self->button_stack));
 	}
-- 
2.30.2

